File: Algorithm/C++/1000+/1534. Count Good Triplets.cc
#include <vector>
#include <cmath>


class Solution {
public:
    int countGoodTriplets(std::vector<int>& arr, int a, int b, int c) {
       int n = arr.size(), result = 0;
       for(int i=0; i < n; i++) {
           for(int j=i+1; j < n; j++) {
               if(std::abs(arr[i] - arr[j]) > a) continue;
               for(int k=j+1; k < n; k++) {
                   if(std::abs(arr[j] - arr[k]) > b || std::abs(arr[i] - arr[k]) > c) continue;
                   result++;
               }
           }
       }
       return result;
    }
};

File: Algorithm/Python/1000+/1534. Count Good Triplets.py
class Solution(object):
    def countGoodTriplets(self, arr, a, b, c):
        """
        :type arr: List[int]
        :type a: int
        :type b: int
        :type c: int
        :rtype: int
        """
        result = 0
        for i in range(len(arr)):
            for j in range(i+1, len(arr)):
                if abs(arr[i] - arr[j]) > a:
                    continue
                for k in range(j+1, len(arr)):
                    if abs(arr[j] - arr[k]) > b or abs(arr[i] - arr[k]) > c:
                        continue
                    result += 1
        return result